src/core/jwk.rs

```diff
@@ -863,6 +1019,41 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_eddsa_verification() {
+        let key_ed25519: CoreJsonWebKey =
+            serde_json::from_str(TEST_ED_PUB_KEY_ED25519).expect("deserialization failed");
+        let pkcs1_signing_input = "eyJhbGciOiJFZDI1NTE5IiwidHlwIjoiSldUIn0.eyJpc3MiOiJqb2UifQ";
+        let signature_ed25519 = "Augr7UH6hUbWVN0PHqSD5U0bb8y9UOw_eef09ZS5d5haUar_qAto8gyLJxUhNF5wHPoXhdvSGowkPvjiKsEsCQ";
+
+        let signature_ed25519_other = "xb4NH-q33sCaRXf1ZhnzQxd4o5ZkBWKd9vGibacqPMAblW_mIJLm9kGerqHX08SPoeDY-dYUmZQz9ls6csfvAw";
+        let signature_ed448 = "xxXVMyaYYePdGfMOdU0nENuc70pKwP3vJuc_jBA0rCW-RtbvBLSsc0D9iCPzhrPmQ2X1nTjPkGiAXJ0_NslDBvy3sHu88N64YhnnYBWwwHttBU0jijn_ikbBUHzUwzGuasRFb1ESG_PwedhEcMi-YAwA";
+
+        // test ed25519
+        verify_signature(
+            &key_ed25519,
+            &CoreJwsSigningAlgorithm::EdDsaEd25519,
+            pkcs1_signing_input,
+            signature_ed25519,
+        );
+
+        // signature from ed448 variant
+        verify_invalid_signature(
+            &key_ed25519,
+            &CoreJwsSigningAlgorithm::EdDsaEd25519,
+            pkcs1_signing_input,
+            signature_ed448,
+        );
+
+        // different signature
+        verify_invalid_signature(
+            &key_ed25519,
+            &CoreJwsSigningAlgorithm::EdDsaEd25519,
+            pkcs1_signing_input,
+            signature_ed25519_other,
+        );
```

Comment:
I think we probably also want to call `JsonWebKey::verify_signature()` with `CoreJwsSigningAlgorithm::EdDsaEd25519` on a non-EdDSA key, similar to: https://github.com/ramosbugs/openidconnect-rs/blob/67474002298c5b818adfcd9f74c898adf42758c6/src/core/jwk.rs#L968-L985